bpf/lib/icmp6.h

```diff
@@ -121,7 +121,7 @@ __section_tail(CILIUM_MAP_CALLS, CILIUM_CALL_SEND_ICMP6_ECHO_REPLY) int tail_icm
 
 	ret = __icmp6_send_echo_reply(skb, nh_off);
 	if (IS_ERR(ret))
-		return send_drop_notify_error(skb, ret, TC_ACT_SHOT);
+		return send_drop_notify_error(skb, ret, TC_ACT_SHOT, DIRECTION_EGRESS); //Check if direction is correct
```

Comment:
For this one, you can trace similar to the overlay case by checking where `CILIUM_CALL_SEND_ICMP6_ECHO_REPLY` is used.

Reply:
It's called in `icmp6_handle` for case `ICMPV6_ECHO_REQUEST` . So I am assuming this should be `DIRECTION_INGRESS` too

Reply:
done. Please check